src/future/future/mod.rs

```diff
@@ -129,6 +133,90 @@ extension_trait! {
         {
             DelayFuture::new(self, dur)
         }
+
+        #[doc = r#"
+            Waits for one of two similarly-typed futures to complete.
+
+            Awaits multiple futures simultaneously, returning all results once complete.
+
+            This function will return a new future which awaits for either one of both
+            futures to complete. If multiple futures are completed at the same time,
+            resolution will occur in the order that they have been passed.
```

Comment:
Do we want to have this guarantee or make it unspecified which result is returned if both features are ready at the same time? My hunch is that "random" could be a better default.

Reply:
Agreed; let's leave it unspecified for now, and actually randomize it later.